lib/v2/parliament/section77.js

```diff
@@ -0,0 +1,191 @@
+const cheerio = require('cheerio');
+const timezone = require('@/utils/timezone');
+
+module.exports = async (ctx) => {
+    const baseUrl = 'https://www.parliament.go.th/section77';
+    const { type = '' } = ctx.params;
+
+    let title = 'ร่างพระราชบัญญัติที่เปิดรับฟังความคิดเห็นตามมาตรา 77 ของรัฐธรรมนูญ';
+
+    if (type) {
+        const [presenter, isMonetaryAct = ''] = type.split('-');
+
+        title +=
+            {
+                presentbymp: 'ที่เสนอโดยสมาชิกสภาผู้แทนราษฏร',
+                presentbyperson: 'ที่เสนอโดยประชาชน',
+                openwsu: 'ที่กำลังเปิดรับฟังความคิดเห็น',
+                closewsu: 'ที่ปิดรับฟังความคิดเห็น',
+                reportwsu: ' รายงานผลการรับฟังความคิดเห็น',
+                substatus1: 'ซึ่งรอคำรับรองจากนายกรัฐมนตรี',
+                substatus2: 'ที่บรรจุเข้าระเบียบวาระ',
+                substatus3: 'ที่พิจารณาแล้ว',
+                closewsubypm: 'ที่นายกฯ ไม่รับรอง',
+            }[presenter] ?? '';
+
+        title +=
+            {
+                m: ' (ประธานสภาผู้แทนราษฎรวินิจฉัยว่า เป็นร่างการเงิน)',
+                nm: ' (ประธานสภาผู้แทนราษฎรวินิจฉัยว่า ไม่เป็นร่างการเงิน)',
+            }[isMonetaryAct] ?? '';
+    }
+
+    const result = {
+        title,
+        link: `${baseUrl}/survey_more_news.php${type ? '?type=' + type : ''}`,
+        language: 'th-th',
+        item: [],
+    };
+
+    // Fetch the news first so we can know how many pages do we need to fetch
+    const browser = await require('@/utils/puppeteer')();
+    const gotByPPT = async (url) => {
+        const page = await browser.newPage();
+        // intercept all requests
+        await page.setRequestInterception(true);
+        // only allow certain types of requests to proceed
+        page.on('request', (request) => {
+            // in this case, we only allow document requests to proceed
+            request.resourceType() === 'document' ? request.continue() : request.abort();
+        });
+        const response = await page.goto(url, {
+            // specify how long to wait for the page to load
+            waitUntil: 'domcontentloaded',
+        });
+        await page.setContent((await response.buffer()).toString('utf8'));
+
+        // retrieve the HTML content of the page
+        const html = await page.content();
+        // close the tab
+        page.close();
+
+        return html;
+    };
```

Comment:
`got` still works fine if you do deal with the anti-crawling mechanism.
For requests without any valid cookies, the site will response with a 302 with cookie `__nxquid` and you have to use that in order to perform any further requests.
Either extract it from the set-cookie header or use CookieJar.